object/slicer/options.go

```diff
@@ -55,16 +58,29 @@ func (x *Options) SetCopiesNumber(copiesNumber uint32) {
 	x.copiesNumber = copiesNumber
 }
 
-// SetPayloadBuffer sets pre-allocated payloadBuffer to be used to object uploading.
-// The payloadBuffer should have length at least MaxObjectSize+1 from NeoFS,
-// otherwise, it does not affect anything.
+// SetPayloadBuffer sets pre-allocated payloadBuffer to be used to object
+// uploading. The payloadBuffer should have length at least
+// [Options.ObjectPayloadLimit] + 1. If [Options.SetPayloadSize] is called, its
+// argument is used as buffer size requirement instead. Any smaller buffer is
+// ignored.
 func (x *Options) SetPayloadBuffer(payloadBuffer []byte) {
 	x.payloadBuffer = payloadBuffer
 }
 
+// SetPayloadSize allows to specify object's payload size known in advance. if
+// the size is fixed, the option is recommended as it improves the performance
+// of the application using the [Slicer].
+func (x *Options) SetPayloadSize(size uint64) {
```

Comment:
How is it different from `SetObjectPayloadLimit`?

Reply:
`SetObjectPayloadLimit` limits size of stored objects (e.g. 64MB), `SetPayloadSize` sets known size of the object (e.g. 1GB). Imo `SetObjectPayloadLimit` is not clear enough naming. I can suggest `SetStoredObjectPayloadLimit` or `SetProducedObjectPayloadLimit`

Reply:
I don't see a lot of difference buffering-wise, I can pass some limited `payloadBuffer` and if it works for the size of the file then I don't need it to be of `Options.ObjectPayloadLimit + 1` size. It can be useful for object metadata though.

Reply:
>  then I don't need it to be of `Options.ObjectPayloadLimit + 1` size

ofc not, it's written in docs